### ISSUE TITLE
Impact of Sample size on B0maf

### ISSUE THREAD

**Issue by hzz0024** [bioXiaoheng/BallerMixPlus#5]
Hi Xiaoheng,

Thanks for building this fantastic program and making it public! I have an observation regarding the B0maf that I want to confirm with you. That is, B0maf values are positively correlated to the total number of alleles observed (i.e., sample size) in the input, right? 

I am asking because when I compare the B0maf values of a few hundred random SNPs (no selection) in two populations with different samples size (e.g., 50 vs. 200), the B0maf values seem to be higher in populations with a bigger sample size. Alternatively, assuming these SNPs are neutral, I would expect to see a similar B0maf distribution between the two populations.

Thanks,
Honggang

**Comment by bioXiaoheng** [comment]
Hi Honggang,

Thank you for reaching out and I'm glad that my codes have been useful!

As for the value of B0maf, have you by chance checked out the number of sites in each window? For larger sample sizes, on average we'd expect to see denser segregating sites. My guess is, in your case, even for the same size (in kb) region, B0maf on a larger sample size tends to include more segregating sites than that for a small sample size. If we approximate a likelihood ratio statistic (one for each SNP) as (asymptotically) following a chi-square dist'n (df=1 here), the composite LR of N sites would be similar in scale to the sum of N chi2-distributed random variables (so df=N). Although the actual empirical distribution isn't chi-square, but at least you can get a sense of scale here.

Hope that helps!
Xiaoheng

**Comment by hzz0024** [comment]
Hi Xiaoheng,

Thank you for your prompt reply! I agree that the composite LR is in scale with the number of sites within a window. In my case, I was comparing a center subpopulation (n=49) to a metapopulation (n=239), assuming a stepping-stone model with high gene flow. I checked the B0maf results and found no significant difference between the number of sites in each window (nSites, p = 0.2). Attached are the two outputs, along with the corresponding inputs and the sfs files. Interestingly, the average B0maf is approximately twice larger in metapopulation (43.61) than in the subpopulation (22.90). I am wondering what else might cause such a big difference. Thanks!

The code that I used for data running is shown below,

```
# for metapopulation
python3 BalLeRMix+_v1.py --noSub --MAF --spect meta1.neutral.sfs.nosub_spect.txt -i meta1_rep1_chr1.mafs.input.txt -o meta1_rep1_chr1.mafs.output.500bp.s1.txt --usePhysPos --fixWinSize -w 500 --step 1
# for subpopulation
python3 BalLeRMix+_v1.py --noSub --MAF --spect pop1.neutral.sfs.nosub_spect.txt -i pop1_rep1_chr1.mafs.input.txt -o pop1_rep1_chr1.mafs.output.500bp.s1.txt --usePhysPos --fixWinSize -w 500 --step 1
```

[B0maf_example.zip](https://github.com/bioXiaoheng/BallerMixPlus/files/9916561/B0maf_example.zip)

**Comment by bioXiaoheng** [comment]
Ah that makes sense now---it's a metapopulation.

When you merge samples from multiple different populations, you introduce population structures in your data, and pop structure elevates genetic diversity. (This also makes another case why we shouldn't use chi-2 distributions for the CLR.)  

Especially for balancing selection, I generally advise against pooling samples especially for balancing selection, as having a high-diversity background conflates inferences.

On a different note---now that I'm more aware that this is the ballermix*plus*, I feel obliged to remind you that the extended B0 and B0maf virtually have no power to detect positive selection at all (see figure S3 in the paper supplement), so please be mindful when you interpret the results.  If your organism does have a sister species's genome sequence (e.g. an .AXT alignment file for the two genomes), I highly recommend you to check out the parsing scripts in this repo and incorporate substitutions in your data.

Best,
Xiaoheng

**Comment by hzz0024** [comment]
Thanks! It makes sense that the exist of population structure will impact the CLR, although the magnitude of changes was larger than I expected. Note here the Fst from pairwise contrasts (among subpopulations) ranges from 0.00025 to 0.00075, as I was focusing on a metapopulation with very high gene flow. 

I am sorry that my initial question is not clarified well. Now I have a new question regarding the relationship between CLR and sample size n. My understanding it that the probability of observing a polymorphic site with k minor alleles out of n sampled alleles is mostly affected by the n, although other parameters may have a impact on final B0maf CLR values. The larger the n, the less likely to observe a intermediate K allele under balancing selection. But if a loci is under balancing selection, for the _**identical population**_, a larger n may lead to higher CLR values across the tested sites. 

In other words, I think the B0maf CLR is a relative measurement of balancing selection, which are comparable for populations with equal size (same as LD r2 calculations). This is just my rough thought, but I'd love to hear if you have any comments or corrections. Again, thanks for your prompt reply and valuable reminder.

Best,
Honggang
